onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py

```diff
@@ -94,6 +136,25 @@ def load_model(
         if version.parse(transformers_version) >= version.parse("4.36.0"):
             extra_kwargs["attn_implementation"] = "eager"
         model = WhisperForConditionalGeneration.from_pretrained(model_name_or_path, cache_dir=cache_dir, **extra_kwargs)
+        if model_impl == "openai":
```

Comment:
Is it possible to refactor this method to eliminate some code duplication? It seems both the OpenAI and Hugging Face paths share some similar code (ex: both appear to use `WhisperDecoder` and `WhisperEncoderDecoderInit`).

To load the PyTorch model, you can use this snippet that was recently added to work with different `transformers` versions.

https://github.com/microsoft/onnxruntime/blob/18c3acb1983e402248365b9f2babfaf8e2182e81/onnxruntime/python/tools/transformers/models/whisper/whisper_helper.py#L93-L96